src/gql/GQLWrapper.tsx

```diff
@@ -40,6 +40,39 @@ interface ClientLinkParams {
 
 const cache = new InMemoryCache({
   typePolicies: {
+    Query: {
+      fields: {
+        projectEvents: {
+          keyArgs: false,
+        },
+        repoEvents: {
+          keyArgs: false,
+        },
+      },
+    },
+    ProjectEvents: {
+      fields: {
+        count: {
+          merge(existing = 0, incoming = 0) {
+            return existing + incoming;
+          },
+        },
+        eventLogEntries: {
+          merge(existing = [], incoming = []) {
+            return [...existing, ...incoming];
+          },
+        },
+      },
+    },
+    ProjectAlias: {
+      keyFields: false,
+    },
+    Project: {
+      keyFields: false,
+    },
+    ProjectSubscription: {
+      keyFields: false,
```

Comment:
You should be able to use some combination of projectIdentifier and some other field for this.

Reply:
The problem with these few fields occurs when Apollo tries to merge the `before` and `after` fields in a `ProjectEventLogEntry`. These structs have no distinguishing field that could be used as a key; in fact they are usually almost exactly the same % whatever modification was made on save. The only field that could serve to distinguish them is whether they are a descendant of `before` or `after`, and Apollo doesn't provide a way to use the key name in in `keyFields`. Thus we have no choice but to pretty much disable caching altogether (which is fine—it really would not be useful in this context).